src/Donations/Components/ContactsForm.tsx

```diff
@@ -260,7 +260,7 @@ function ContactsForm(): ReactElement {
                   label={t("email")}
                   variant="outlined"
                   data-test-id="test-email"
-                  disabled={isAuthenticated}
+                  disabled={profile?.email.length !== 0}
```

Comment:
This logic does not work for guest users. The email field is always disabled. 

For logged in/proxy-logged in users, there is a `profile` with an `email`, So the statement `profile?.email.length !== 0` evaluates to `true`.

For guest users, `profile` is `null`. So the statement `profile?.email.length !== 0` evaluates to `true`.